processing/src/main/java/org/apache/druid/segment/VirtualColumn.java

```diff
@@ -299,13 +300,16 @@
 
   /**
    * Get the {@link ColumnIndexSupplier} for the specified virtual column, with the assistance of a
-   * {@link ColumnSelector} to allow reading things from segments. If the virtual column has no indexes, this method
-   * will return null, or may also return a non-null supplier whose methods may return null values - having a supplier
-   * is no guarantee that the column has indexes.
+   * {@link ColumnIndexSelector} to allow reading things from segments. If the virtual column has no indexes, this
+   * method will return null, or may also return a non-null supplier whose methods may return null values - having a
+   * supplier is no guarantee that the column has indexes.
    */
   @SuppressWarnings("unused")
   @Nullable
-  default ColumnIndexSupplier getIndexSupplier(String columnName, ColumnSelector columnSelector)
+  default ColumnIndexSupplier getIndexSupplier(
+      String columnName,
```

Comment:
I added `columnName` parameter to this method originally just in case stuff was using the 'usesDotNotation', unsure if the column name would be useful in such a situation, though perhaps it isn't really necessary and could be removed...

Reply:
There hasn't been much use that comes up for dot notation other than the original `map` virtual column. The JSON stuff could have used it, but we ended up going the more standard `JSON_VALUE` way instead. IMO it's fine to say simplify this and say that dot-notation virtual columns don't get to supply indexes. I'm OK either way though really.